### PR TITLE
Add proper support for Rust.

### LINKED ISSUE #347: Rust highlighting doesn't handle lifetime parameters correctly
```
What steps will reproduce the problem?

The highlighting for Rust does not properly handle lifetimes. It treats the 
lifetime sigil ' as the start of a string instead (even though Rust doesn't 
support single-quoted strings). This causes bad rendering.

This causes a problem on StackOverflow, when people post questions about Rust. 
For example, in a question asked 19 hours ago 
(http://stackoverflow.com/questions/24855830/how-convert-a-string-in-str-in-impl
-block-in-rust), the first code block contains a struct definition with a field 
`title: &'static str`. The rest of the code block from the ' is colored red, as 
if it were a string, when in fact 'static is a lifetime specifier.

In Rust, the ' token is either used to start a character literal, or a 
lifetime. if the ' is followed by a single unicode character (or a single 
escape, such as \n, \t, \u2022), and then another ', then it is a character 
literal. Otherwise, if it's followed by a single identifier that meets the 
identifier rules, the ' with the identifier are considered a lifetime 
(otherwise, it's assumed to be an unterminated character literal).
```

Original issue reported on code.google.com by `kball...@gmail.com` on 21 Jul 2014 at 6:35

### PR DESCRIPTION
**Pull request by chris-morgan**
Closes #347.

### CODE REVIEW

File: src/lang-rust.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+ */
+
+PR['registerLangHandler'](
+    PR['createSimpleLexer']([], [
+		// Whitespace
+		[PR['PR_PLAIN'],       /^[\t\n\r \xA0]+/],
+
+		// Single line comments
+		[PR['PR_COMMENT'], /^\/\/.*/],
+		// Block comments (sadly I do not see how to make this cope with comment nesting as it should)
```
Comment (mikesamuel) [inline src/lang-rust.js]: Yep.   Sorry.  I haven't had time to relax the regular-lexical-grammar assumptions, and there's no workaround for Rust&Ocaml's nesting comments the way there is for Lua.

Comment (mikesamuel) [inline src/lang-rust.js]: Yep.   Sorry.  I haven't had time to relax the regular-lexical-grammar assumptions, and there's no workaround for Rust&Ocaml's nesting comments the way there is for Lua.

**Comment by mikesamuel** [comment]
I'm horribly ignorant of typical code-review protocol on github.   There's a branch conflict.  Is it typical for you to merge and push or does the committer maintainer typically merge and commit?

**Comment by chris-morgan** [comment]
@mikesamuel I’ve rebased to fix the conflict.

**Comment by sfreilich** [comment]
This change breaks tests. Specifically, in the "rust" test in tests/prettify_test.html it changes the class of this span from atn (PR_ATTRIB_NAME) to atv (PR_ATTRIB_VALUE):
<span class="atv">#![feature(code_prettification)]</span>

Plus maybe other things (the diff is large, but I can't spot other changes just eyeballing it). I'd write a fix, but I'm not sure if the change is intended or not.

**Comment by sfreilich** [comment]
There are some other broken tests, but those seem to be the result of d0fd4e3.

**Comment by sfreilich** [comment]
Test fix in PR #409.

**Pull request merged**